### PR TITLE
init refarch-frontend

### PR DESCRIPTION
**Pull request by DanielOber**
**Description**

- removed api-gateway
- moved .gitattributes into the frontend
- added Mucatar_URL alls enviornment variable

**Reference**

Issues #56

### CODE REVIEW

File: refarch-frontend/vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    ViteFonts({
+      google: {
+        families: [
+          {
+            name: "Roboto",
+            styles: "wght@100;300;400;500;700;900",
+          },
+        ],
+      },
+    }),
```
Comment (devtobi) [inline refarch-frontend/vite.config.ts:29]: I wonder that we explicitly "import" Roboto font here. Where are the font getting pulled from? Are they included in the final build or loaded externally from Google servers? Some browsers might have them integrated out of the box, but others don't. We shouldn't force loading from external servers and make sure fonts are always included. On top of that, why do we need to define font here? Isn't this Vuetify default?

File: refarch-frontend/tests/unit/example.spec.ts
Comment (devtobi) [inline refarch-frontend/tests/unit/example.spec.ts:1]: Are we going to support component testing in the future? There was not that much consent for unit testing Vue components in the past. We should discuss if we want to further include this.

File: refarch-frontend/vitest.config.ts
Comment (devtobi) [inline refarch-frontend/vitest.config.ts:1]: Might not be needed anymore if we remove unit test support. Should be discussed.

File: refarch-frontend/src/views/HomeView.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+</script>
+
+<style scoped>
+.UP {
+  color: limegreen;
+}
+
+.DOWN {
+  color: lightcoral;
+}
```
Comment (devtobi) [inline refarch-frontend/src/views/HomeView.vue:51]: Naming CSS classes like this is generally bad practice (see BEM notation). We should consider refactoring this.

File: refarch-frontend/src/views/GetStartedView.vue
Comment (devtobi) [inline refarch-frontend/src/views/GetStartedView.vue:1]: Since here only internal links are included, we should consider removing those and re-adding those once we have useful public documentation.

File: refarch-frontend/src/types/itm-appswitcher-vue/itm-appswitcher-vue.d.ts
Comment (devtobi) [inline refarch-frontend/src/types/itm-appswitcher-vue/itm-appswitcher-vue.d.ts:1]: Why is this needed? Isn't the dependency correctly typed? (That's the main purpose for .d.ts files). I suggest trying to remove the file and if it does not work at least collecting type definitions from external libraries in a `declarations.d.ts` file at `src`-root.

File: refarch-frontend/src/types/User.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  u.username = "Local Development User";
+  u.displayName = "Local Development User";
+  u.authorities = [
+    // todo add authorities
+  ];
+  u.user_roles = [
+    // todo add user roles
+  ];
+  return u;
+}
```
Comment (devtobi) [inline refarch-frontend/src/types/User.ts:31]: I think we should refactor this to a static class function instead of a separate function.

File: refarch-frontend/src/stores/user.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { defineStore } from "pinia";
+import { computed, ref } from "vue";
+
+import User from "@/types/User";
+
+export interface UserState {
+  user: User | null;
```
Comment (devtobi) [inline refarch-frontend/src/stores/user.ts:7]: General topic: `null` handling is kind of irregular in different places and for me union types like `MyType | null` look not very clean. Maybe there is a more TypeScript-native like way to do things?
Sometimes we have `* | null`, sometimes we have `* | undefined`. We should find a consistent solution.

File: refarch-frontend/src/plugins/vuetify.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    themes: {
+      light: {
+        colors: {
+          primary: "#333333",
+          secondary: "#FFCC00",
+          accent: "#7BA4D9",
+          success: "#69BE28",
+          error: "#FF0000",
+        },
+      },
```
Comment (devtobi) [inline refarch-frontend/src/plugins/vuetify.ts:17]: We could think about separating theme definition into separate file and then import here. Might make the whole theming stuff more clear especially when we add things like dark mode, high contrast mode etc.

File: refarch-frontend/src/plugins/index.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import type { App } from "vue";
+
+import router from "../router";
+import pinia from "./pinia";
+import vuetify from "./vuetify";
```
Comment (devtobi) [inline refarch-frontend/src/plugins/index.ts:5]: We should always import files using the `@/..` shorthand provided by Vite. This should also be possible in non Vue SFCs.

File: refarch-frontend/src/main.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { createApp } from "vue";
+
+import { registerPlugins } from "@/plugins";
+import App from "./App.vue";
```
Comment (devtobi) [inline refarch-frontend/src/main.ts:4]: We should always import files using the `@/..` shorthand provided by Vite. This should also be possible in non Vue SFCs.

File: refarch-frontend/src/composables/formatter.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+      ? text.toLowerCase().replace(/^\w/, (c) => c.toUpperCase())
+      : "";
+  }
+
+  function formatDate(date: Date): string {
+    return date ? moment(date).format("L") : "";
+  }
+
+  return { startingCharUpperCase, formatDate };
+}
```
Comment (devtobi) [inline refarch-frontend/src/composables/formatter.ts:15]: This file tries to mimic a Vue composable to expose the mentioned methods. However no Vue specific stuff like Refs etc. is used here so this can be used totally outside the Vue "ecosystem". That's why I suggest to refactor and rename this file (e.g. a static class) to indicate its not Vue dependent.

File: refarch-frontend/src/components/common/DatetimeInput.vue
Comment (devtobi) [inline refarch-frontend/src/components/common/DatetimeInput.vue:1]: Is there no Vuetify native way of combining date with time input?

**Review by devtobi — CHANGES_REQUESTED** [review]
Review done!

File: refarch-frontend/src/App.vue [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+          cols="6"
+          class="d-flex align-center justify-center"
+        >
+          <v-text-field
+            id="suchfeld"
+            v-model="query"
+            flat
+            variant="solo-inverted"
+            hide-details
+            label="Suche"
```
[the remark this answers is not in the document]
Comment (DanielOber) [reply refarch-frontend/src/App.vue:29]: No, the UI itself should still be in german

File: refarch-frontend/.browserslistrc
[the remark this answers is not in the document]
Comment (DanielOber) [reply refarch-frontend/.browserslistrc:1]: #66

File: refarch-frontend/.env.development [hunk 10]
```diff
@@ -0,0 +1,2 @@
+VITE_VUE_APP_API_URL=http://localhost:8082
```
[the remark this answers is not in the document]
Comment (DanielOber) [reply refarch-frontend/.env.development:1]: #67

File: refarch-frontend/src/App.vue [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+        <v-col
+          cols="3"
+          class="d-flex align-center justify-start"
+        >
+          <v-app-bar-nav-icon @click.stop="drawer = !drawer" />
+          <router-link to="/">
+            <v-toolbar-title class="font-weight-bold">
+              <span class="text-white">RefArch-</span>
+              <span class="text-secondary">Kick</span>
+              <span class="text-white">Starter</span>
```
[the remark this answers is not in the document]
Comment (DanielOber) [reply refarch-frontend/src/App.vue:15]: #68

File: refarch-frontend/src/App.vue [hunk 12]
```diff
@@ -0,0 +1,139 @@
+<template>
+  <v-app>
+    <the-snackbar />
+    <v-app-bar color="primary">
+      <v-row align="center">
+        <v-col
+          cols="3"
+          class="d-flex align-center justify-start"
+        >
+          <v-app-bar-nav-icon @click.stop="drawer = !drawer" />
```
[the remark this answers is not in the document]
Comment (DanielOber) [reply refarch-frontend/src/App.vue:10]: #69

File: refarch-frontend/src/App.vue [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+            v-if="appswitcherBaseUrl"
+            :base-url="appswitcherBaseUrl"
+            :tags="['global']"
+          />
+          <v-btn
+            variant="text"
+            icon
+          >
+            <lhm-avatar
+              v-if="userStore.getUser !== null"
```
[the remark this answers is not in the document]
Comment (DanielOber) [reply refarch-frontend/src/App.vue:50]: #71

File: refarch-frontend/src/api/FetchUtils.ts
[the remark this answers is not in the document]
Comment (DanielOber) [reply refarch-frontend/src/api/FetchUtils.ts:1]: #72

File: refarch-frontend/src/components/TheSnackbar.vue [hunk 14]
```diff
@@ -0,0 +1,63 @@
+<template>
+  <v-snackbar
+    id="snackbar"
+    v-model="show"
+    :color="color"
+    :timeout="timeout"
+  >
+    {{ message }}
+    <v-btn
+      v-if="color === 'error'"
```
[the remark this answers is not in the document]
Comment (DanielOber) [reply refarch-frontend/src/components/TheSnackbar.vue:10]: #74

File: refarch-frontend/src/views/HomeView.vue (hunk 2 above)
Comment (DanielOber) [reply refarch-frontend/src/views/HomeView.vue:51]: #76

File: refarch-frontend/vitest.config.ts
Comment (DanielOber) [reply refarch-frontend/vitest.config.ts:1]: #77

File: refarch-frontend/src/views/GetStartedView.vue
Comment (DanielOber) [reply refarch-frontend/src/views/GetStartedView.vue:1]: #78

File: refarch-frontend/src/plugins/vuetify.ts (hunk 5 above)
Comment (DanielOber) [reply refarch-frontend/src/plugins/vuetify.ts:17]: not sure why we would do this. If you want to change the vuetify theme, you need to know where you have to change it and that should be in the vuetify documentation. Moving thing away from it doesnt sound like a good idea.

File: refarch-frontend/src/plugins/index.ts (hunk 6 above)
Comment (DanielOber) [reply refarch-frontend/src/plugins/index.ts:5]: #79

File: refarch-frontend/src/main.ts (hunk 7 above)
Comment (DanielOber) [reply refarch-frontend/src/main.ts:4]: #79

**Review by devtobi — APPROVED** [review]
Moved all conversations to separate issues. LGTM now!
